// @sqds/multisig/lib/generated/instructions/configTransactionExecute.js
/**
 * This code was GENERATED using the solita package.
 * Please DO NOT EDIT THIS FILE, instead rerun solita to update it or write a wrapper to add functionality.
 *
 * See: https://github.com/metaplex-foundation/solita
 */
import * as beet from '@metaplex-foundation/beet';
import * as web3 from '@solana/web3.js';
/**
 * @category Instructions
 * @category ConfigTransactionExecute
 * @category generated
 */
export const configTransactionExecuteStruct = new beet.BeetArgsStruct([['instructionDiscriminator', beet.uniformFixedSizeArray(beet.u8, 8)]], 'ConfigTransactionExecuteInstructionArgs');
export const configTransactionExecuteInstructionDiscriminator = [
    114, 146, 244, 189, 252, 140, 36, 40,
];
/**
 * Creates a _ConfigTransactionExecute_ instruction.
 *
 * Optional accounts that are not provided default to the program ID since
 * this was indicated in the IDL from which this instruction was generated.
 *
 * @param accounts that will be accessed while the instruction is processed
 * @category Instructions
 * @category ConfigTransactionExecute
 * @category generated
 */
export function createConfigTransactionExecuteInstruction(accounts, programId = new web3.PublicKey('SQDS4ep65T869zMMBKyuUq6aD6EgTu8psMjkvj52pCf')) {
    const [data] = configTransactionExecuteStruct.serialize({
        instructionDiscriminator: configTransactionExecuteInstructionDiscriminator,
    });
    const keys = [
        {
            pubkey: accounts.multisig,
            isWritable: true,
            isSigner: false,
        },
        {
            pubkey: accounts.member,
            isWritable: false,
            isSigner: true,
        },
        {
            pubkey: accounts.proposal,
            isWritable: true,
            isSigner: false,
        },
        {
            pubkey: accounts.transaction,
            isWritable: true,
            isSigner: false,
        },
        {
            pubkey: accounts.rentPayer ?? programId,
            isWritable: accounts.rentPayer != null,
            isSigner: accounts.rentPayer != null,
        },
        {
            pubkey: accounts.systemProgram ?? programId,
            isWritable: false,
            isSigner: false,
        },
    ];
    if (accounts.anchorRemainingAccounts != null) {
        for (const acc of accounts.anchorRemainingAccounts) {
            keys.push(acc);
        }
    }
    const ix = new web3.TransactionInstruction({
        programId,
        keys,
        data,
    });
    return ix;
}
