// @sqds/multisig/lib/generated/instructions/multisigChangeThreshold.js
/**
 * This code was GENERATED using the solita package.
 * Please DO NOT EDIT THIS FILE, instead rerun solita to update it or write a wrapper to add functionality.
 *
 * See: https://github.com/metaplex-foundation/solita
 */
import * as beet from '@metaplex-foundation/beet';
import * as web3 from '@solana/web3.js';
import { multisigChangeThresholdArgsBeet, } from '../types/MultisigChangeThresholdArgs';
/**
 * @category Instructions
 * @category MultisigChangeThreshold
 * @category generated
 */
export const multisigChangeThresholdStruct = new beet.FixableBeetArgsStruct([
    ['instructionDiscriminator', beet.uniformFixedSizeArray(beet.u8, 8)],
    ['args', multisigChangeThresholdArgsBeet],
], 'MultisigChangeThresholdInstructionArgs');
export const multisigChangeThresholdInstructionDiscriminator = [
    141, 42, 15, 126, 169, 92, 62, 181,
];
/**
 * Creates a _MultisigChangeThreshold_ instruction.
 *
 * Optional accounts that are not provided default to the program ID since
 * this was indicated in the IDL from which this instruction was generated.
 *
 * @param accounts that will be accessed while the instruction is processed
 * @param args to provide as instruction data to the program
 *
 * @category Instructions
 * @category MultisigChangeThreshold
 * @category generated
 */
export function createMultisigChangeThresholdInstruction(accounts, args, programId = new web3.PublicKey('SQDS4ep65T869zMMBKyuUq6aD6EgTu8psMjkvj52pCf')) {
    const [data] = multisigChangeThresholdStruct.serialize({
        instructionDiscriminator: multisigChangeThresholdInstructionDiscriminator,
        ...args,
    });
    const keys = [
        {
            pubkey: accounts.multisig,
            isWritable: true,
            isSigner: false,
        },
        {
            pubkey: accounts.configAuthority,
            isWritable: false,
            isSigner: true,
        },
        {
            pubkey: accounts.rentPayer ?? programId,
            isWritable: accounts.rentPayer != null,
            isSigner: accounts.rentPayer != null,
        },
        {
            pubkey: accounts.systemProgram ?? programId,
            isWritable: false,
            isSigner: false,
        },
    ];
    if (accounts.anchorRemainingAccounts != null) {
        for (const acc of accounts.anchorRemainingAccounts) {
            keys.push(acc);
        }
    }
    const ix = new web3.TransactionInstruction({
        programId,
        keys,
        data,
    });
    return ix;
}
