// @sqds/multisig/lib/generated/instructions/programConfigInit.js
/**
 * This code was GENERATED using the solita package.
 * Please DO NOT EDIT THIS FILE, instead rerun solita to update it or write a wrapper to add functionality.
 *
 * See: https://github.com/metaplex-foundation/solita
 */
import * as beet from '@metaplex-foundation/beet';
import * as web3 from '@solana/web3.js';
import { programConfigInitArgsBeet, } from '../types/ProgramConfigInitArgs';
/**
 * @category Instructions
 * @category ProgramConfigInit
 * @category generated
 */
export const programConfigInitStruct = new beet.BeetArgsStruct([
    ['instructionDiscriminator', beet.uniformFixedSizeArray(beet.u8, 8)],
    ['args', programConfigInitArgsBeet],
], 'ProgramConfigInitInstructionArgs');
export const programConfigInitInstructionDiscriminator = [
    184, 188, 198, 195, 205, 124, 117, 216,
];
/**
 * Creates a _ProgramConfigInit_ instruction.
 *
 * @param accounts that will be accessed while the instruction is processed
 * @param args to provide as instruction data to the program
 *
 * @category Instructions
 * @category ProgramConfigInit
 * @category generated
 */
export function createProgramConfigInitInstruction(accounts, args, programId = new web3.PublicKey('SQDS4ep65T869zMMBKyuUq6aD6EgTu8psMjkvj52pCf')) {
    const [data] = programConfigInitStruct.serialize({
        instructionDiscriminator: programConfigInitInstructionDiscriminator,
        ...args,
    });
    const keys = [
        {
            pubkey: accounts.programConfig,
            isWritable: true,
            isSigner: false,
        },
        {
            pubkey: accounts.initializer,
            isWritable: true,
            isSigner: true,
        },
        {
            pubkey: accounts.systemProgram ?? web3.SystemProgram.programId,
            isWritable: false,
            isSigner: false,
        },
    ];
    if (accounts.anchorRemainingAccounts != null) {
        for (const acc of accounts.anchorRemainingAccounts) {
            keys.push(acc);
        }
    }
    const ix = new web3.TransactionInstruction({
        programId,
        keys,
        data,
    });
    return ix;
}
