// @sqds/multisig/lib/generated/instructions/spendingLimitUse.js
/**
 * This code was GENERATED using the solita package.
 * Please DO NOT EDIT THIS FILE, instead rerun solita to update it or write a wrapper to add functionality.
 *
 * See: https://github.com/metaplex-foundation/solita
 */
import * as beet from '@metaplex-foundation/beet';
import * as web3 from '@solana/web3.js';
import { spendingLimitUseArgsBeet, } from '../types/SpendingLimitUseArgs';
/**
 * @category Instructions
 * @category SpendingLimitUse
 * @category generated
 */
export const spendingLimitUseStruct = new beet.FixableBeetArgsStruct([
    ['instructionDiscriminator', beet.uniformFixedSizeArray(beet.u8, 8)],
    ['args', spendingLimitUseArgsBeet],
], 'SpendingLimitUseInstructionArgs');
export const spendingLimitUseInstructionDiscriminator = [
    16, 57, 130, 127, 193, 20, 155, 134,
];
/**
 * Creates a _SpendingLimitUse_ instruction.
 *
 * Optional accounts that are not provided default to the program ID since
 * this was indicated in the IDL from which this instruction was generated.
 *
 * @param accounts that will be accessed while the instruction is processed
 * @param args to provide as instruction data to the program
 *
 * @category Instructions
 * @category SpendingLimitUse
 * @category generated
 */
export function createSpendingLimitUseInstruction(accounts, args, programId = new web3.PublicKey('SQDS4ep65T869zMMBKyuUq6aD6EgTu8psMjkvj52pCf')) {
    const [data] = spendingLimitUseStruct.serialize({
        instructionDiscriminator: spendingLimitUseInstructionDiscriminator,
        ...args,
    });
    const keys = [
        {
            pubkey: accounts.multisig,
            isWritable: true,
            isSigner: false,
        },
        {
            pubkey: accounts.member,
            isWritable: false,
            isSigner: true,
        },
        {
            pubkey: accounts.spendingLimit,
            isWritable: true,
            isSigner: false,
        },
        {
            pubkey: accounts.vault,
            isWritable: true,
            isSigner: false,
        },
        {
            pubkey: accounts.destination,
            isWritable: true,
            isSigner: false,
        },
        {
            pubkey: accounts.systemProgram ?? programId,
            isWritable: false,
            isSigner: false,
        },
        {
            pubkey: accounts.mint ?? programId,
            isWritable: false,
            isSigner: false,
        },
        {
            pubkey: accounts.vaultTokenAccount ?? programId,
            isWritable: accounts.vaultTokenAccount != null,
            isSigner: false,
        },
        {
            pubkey: accounts.destinationTokenAccount ?? programId,
            isWritable: accounts.destinationTokenAccount != null,
            isSigner: false,
        },
        {
            pubkey: accounts.tokenProgram ?? programId,
            isWritable: false,
            isSigner: false,
        },
    ];
    if (accounts.anchorRemainingAccounts != null) {
        for (const acc of accounts.anchorRemainingAccounts) {
            keys.push(acc);
        }
    }
    const ix = new web3.TransactionInstruction({
        programId,
        keys,
        data,
    });
    return ix;
}
