// @sqds/multisig/lib/instructions/configTransactionExecute.js
import { SystemProgram } from "@solana/web3.js";
import { createConfigTransactionExecuteInstruction, PROGRAM_ID, } from "../generated";
import { getProposalPda, getTransactionPda } from "../pda";
export function configTransactionExecute({ multisigPda, transactionIndex, member, rentPayer, spendingLimits, programId = PROGRAM_ID, }) {
    const [proposalPda] = getProposalPda({
        multisigPda,
        transactionIndex,
        programId,
    });
    const [transactionPda] = getTransactionPda({
        multisigPda,
        index: transactionIndex,
        programId,
    });
    return createConfigTransactionExecuteInstruction({
        multisig: multisigPda,
        member,
        proposal: proposalPda,
        transaction: transactionPda,
        rentPayer,
        systemProgram: SystemProgram.programId,
        anchorRemainingAccounts: spendingLimits?.map((spendingLimit) => ({
            pubkey: spendingLimit,
            isWritable: true,
            isSigner: false,
        })),
    }, programId);
}
